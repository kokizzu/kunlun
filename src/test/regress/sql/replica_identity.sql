--DDL_STATEMENT_BEGIN--
CREATE TABLE test_replica_identity (
       id serial primary key,
       keya varchar(50) not null,
       keyb varchar(50) not null,
       nonkey  varchar(50),
       CONSTRAINT test_replica_identity_unique_defer UNIQUE (keya, keyb) DEFERRABLE,
       CONSTRAINT test_replica_identity_unique_nondefer UNIQUE (keya, keyb)
);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE TABLE test_replica_identity_othertable (id serial primary key);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE INDEX test_replica_identity_keyab ON test_replica_identity (keya, keyb);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE UNIQUE INDEX test_replica_identity_keyab_key ON test_replica_identity (keya, keyb);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE UNIQUE INDEX test_replica_identity_nonkey ON test_replica_identity (keya, nonkey);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE INDEX test_replica_identity_hash ON test_replica_identity USING hash (nonkey);
--DDL_STATEMENT_END--
CREATE UNIQUE INDEX test_replica_identity_expr ON test_replica_identity (keya, keyb, (3));
CREATE UNIQUE INDEX test_replica_identity_partial ON test_replica_identity (keya, keyb) WHERE keyb != '3';

-- default is 'd'/DEFAULT for user created tables
SELECT relreplident FROM pg_class WHERE oid = 'test_replica_identity'::regclass;
-- but 'none' for system tables
SELECT relreplident FROM pg_class WHERE oid = 'pg_class'::regclass;
SELECT relreplident FROM pg_class WHERE oid = 'pg_constraint'::regclass;

----
-- Make sure we detect ineligible indexes
----

-- Make sure index cases succeed
----

--DDL_STATEMENT_BEGIN--
DROP TABLE test_replica_identity;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP TABLE test_replica_identity_othertable;
--DDL_STATEMENT_END--