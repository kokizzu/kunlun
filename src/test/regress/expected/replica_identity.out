CREATE TABLE test_replica_identity (
       id serial primary key,
       keya varchar(50) not null,
       keyb varchar(50) not null,
       nonkey  varchar(50),
       CONSTRAINT test_replica_identity_unique_defer UNIQUE (keya, keyb) DEFERRABLE,
       CONSTRAINT test_replica_identity_unique_nondefer UNIQUE (keya, keyb)
);
CREATE TABLE
CREATE TABLE test_replica_identity_othertable (id serial primary key);
CREATE TABLE
CREATE INDEX test_replica_identity_keyab ON test_replica_identity (keya, keyb);
CREATE INDEX
CREATE UNIQUE INDEX test_replica_identity_keyab_key ON test_replica_identity (keya, keyb);
CREATE INDEX
CREATE UNIQUE INDEX test_replica_identity_nonkey ON test_replica_identity (keya, nonkey);
CREATE INDEX
CREATE INDEX test_replica_identity_hash ON test_replica_identity USING hash (nonkey);
CREATE INDEX
CREATE UNIQUE INDEX test_replica_identity_expr ON test_replica_identity (keya, keyb, (3));
CREATE INDEX
CREATE UNIQUE INDEX test_replica_identity_partial ON test_replica_identity (keya, keyb) WHERE keyb != '3';
psql:sql/replica_identity.sql:17: ERROR:  partial index isn't supported for remote relations
-- default is 'd'/DEFAULT for user created tables
SELECT relreplident FROM pg_class WHERE oid = 'test_replica_identity'::regclass;
 relreplident 
--------------
 d
(1 row)

-- but 'none' for system tables
SELECT relreplident FROM pg_class WHERE oid = 'pg_class'::regclass;
 relreplident 
--------------
 n
(1 row)

SELECT relreplident FROM pg_class WHERE oid = 'pg_constraint'::regclass;
 relreplident 
--------------
 n
(1 row)

----
-- Make sure we detect ineligible indexes
----
-- Make sure index cases succeed
----
DROP TABLE test_replica_identity;
DROP TABLE
DROP TABLE test_replica_identity_othertable;
DROP TABLE
